frontend/src/metabase-lib/lib/queries/StructuredQuery.ts

```diff
@@ -330,28 +330,31 @@ class StructuredQueryInner extends AtomicQuery {
     const sourceQuery = this.sourceQuery();
 
     if (sourceQuery) {
+      const fields = sourceQuery.columns().map(column => {
+        const id = [
+          "field",
+          column.name,
+          {
+            "base-type": column.base_type,
+          },
+        ];
+
+        return new Field({
+          ...column,
+          // TODO FIXME -- Do NOT use field-literal unless you're referring to a native query
```

Comment:
Should this TODO get fixed before merging this, or rather: how much has this been hurting us?

Reply:
This TODO is from aeons ago. I'll keep it for now, but may remove soon as not sure what the value is.